### ISSUE TITLE
router: do not finish bootstrap until all the routes are resolved (#14327) (541de26), closes #12162

### ISSUE THREAD

**Issue by yyaman** [angular/angular#14701]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
```

**Current behavior**
After upgrading to 2.4.8, route resolution is starting at the same time as APP_INITIALIZER. I understand that bootstrap should not be completed before the route resolution, but shouldn't the APP_INITIALIZER finish first before starting the route resolution? Our application depends on APP_INITIALIZER provider to load settings first. These settings are used in the the route resolves. After 2.4.8, this is broken.

**Expected behavior**
APP_INITIALIZER Promise should be resolved before starting the route resolutions.

**Minimal reproduction of the problem with instructions**
export const ConfigServiceProvider: FactoryProvider = {
    provide: APP_INITIALIZER,
    useFactory: (config: ConfigService) => () => config.load(),
    deps: [ConfigService],
    multi: true
}

@Injectable()
export class ConfigService {
    public load(): Promise<any> {
        return new Promise((resolve: (value?: any | PromiseLike<any>) => void, reject) => {
            this.http.get("public/appsettings.json").map(res => res.json())
                .subscribe(data => {
                    //populate configuration class that stores language url.
                    resolve(true);
                });
        }
    }
}

@Injectable()
export class LanguageServiceResolve implements Resolve<{ [key: string]: string }> {
    resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<{ [key: string]: string }> | { [key: string]: string } {
        return this.http.get("(url set above)").toPromise().then((x: { [key: string]: string }) => {
            //etc...
        }
        //breaks above because url has not been set yet.
    }
}

**What is the motivation / use case for changing the behavior?**
Be able to load app settings that can be used in route resolution.

**Please tell us about your environment:**
Webpack

* **Angular version:** 2.4.8

* **Browser:** [all ]
 
* **Language:** [TypeScript 2.0.10|  ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by DzmitryShylovich** [comment]
Duplicate of https://github.com/angular/angular/issues/14588

The commit will be reverted in 2.x branch

**Comment by DzmitryShylovich** [comment]
@yyaman fixed by https://github.com/angular/angular/pull/14681
close the issue please
